app/bg/dbs/settings.js

```diff
@@ -57,10 +63,12 @@ export const once = events.once.bind(events)
 
 /**
  * @param {string} key
- * @param {string | number} value
+ * @param {string | number | object} value
  * @returns {Promise<void>}
  */
 export async function set (key, value) {
+  if (['default_search_engines', 'selected_search_engine'].includes(key))
```

Comment:
It's a good idea to move the list of keys that need to be JSONified to a constant at the top of the file, something like `JSON_ENCODED_SETTINGS` so that the lists dont fall out of sync in the various parts of the code